server/src/main/java/org/apache/iotdb/db/engine/upgrade/UpgradeTask.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.db.engine.upgrade;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import org.apache.iotdb.db.engine.storagegroup.TsFileResource;
+import org.apache.iotdb.db.service.UpgradeSevice;
+import org.apache.iotdb.db.utils.UpgradeUtils;
+import org.apache.iotdb.tsfile.fileSystem.FSFactoryProducer;
+import org.apache.iotdb.tsfile.tool.upgrade.UpgradeTool;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class UpgradeTask implements Runnable {
+
+  private final TsFileResource upgradeResource;
+  private static final Logger logger = LoggerFactory.getLogger(UpgradeTask.class);
+  private static final String DOT_SEPERATOR = ",";
+
+  public UpgradeTask(TsFileResource upgradeResource) {
+    this.upgradeResource = upgradeResource;
+  }
+
+
+  private static boolean writeUpgradeLogFile(RandomAccessFile upgradeLog, String content,
+      long position) {
+    UpgradeUtils.getUpgradeLogLock().writeLock().lock();
+    try {
+      upgradeLog.seek(position);
+      upgradeLog.write(content.getBytes());
+      upgradeLog.write(System.getProperty("line.separator").getBytes());
+      return true;
+    } catch (IOException e) {
+      logger.error("write upgrade log file failed, the log file:{}", upgradeLog);
+      return false;
+    } finally {
+      UpgradeUtils.getUpgradeLogLock().writeLock().unlock();
+    }
+  }
+
+  private static long getUpgradeLogLength(RandomAccessFile upgradeLog) {
+    UpgradeUtils.getUpgradeLogLock().readLock().lock();
+    try {
+      return upgradeLog.length();
+    } catch (IOException e) {
+      logger.error("read upgrade log file failed, the log file:{}", upgradeLog);
+      return 0;
+    } finally {
+      UpgradeUtils.getUpgradeLogLock().readLock().unlock();
+    }
+  }
+
+  @Override
+  public void run() {
```

Comment:
Almost all your methods lack `Javadoc`. Please explain the function and important steps of the method in Javadoc for other developers to understand easily.

Reply:
yeah,I have added some important Javadoc.